redbot/cogs/audio/core/events/red.py

```diff
@@ -19,3 +20,34 @@ async def on_red_api_tokens_update(
             self.api_interface.spotify_api.update_token(api_tokens)
         elif service_name == "audiodb":
             self.api_interface.global_cache_api.update_token(api_tokens)
+
+    async def red_delete_data_for_user(
```

Comment:
Some of data deletion (`self.playlist_api`) in Audio cog depends on the cog being ready, so this most likely needs to wait for that event first.